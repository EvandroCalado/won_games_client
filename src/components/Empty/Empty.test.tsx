import { render, screen } from '@testing-library/react';
import { Empty } from '.';

const props = {
  title: 'A simple title',
  description: 'Wanna play?',
  hasLink: true,
};

describe('Empty', () => {
  it('should render default', () => {
    render(<Empty {...props} />);

    expect(
      screen.getByRole('img', { name: /a gamer in a couch playng videogame/i }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { name: /a simple title/i }),
    ).toBeInTheDocument();
    expect(screen.getByText(/wanna play?/i)).toBeInTheDocument();
    expect(
      screen.getByRole('link', { name: /go back to store/i }),
    ).toHaveAttribute('href', '/');
  });

  it('should render without link when hasLink is false', () => {
    render(<Empty {...props} hasLink={false} />);

    expect(
      screen.queryByRole('link', { name: /go back to store/i }),
    ).not.toBeInTheDocument();
  });

  it('should render a snapshot', () => {
    const { container } = render(<Empty {...props} hasLink={false} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
