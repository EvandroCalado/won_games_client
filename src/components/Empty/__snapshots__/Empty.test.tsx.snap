// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty should render a snapshot 1`] = `
<div
  class="flex flex-col items-center justify-center gap-4"
>
  <img
    alt="A gamer in a couch playng videogame"
    data-nimg="1"
    decoding="async"
    fetchpriority="high"
    height="300"
    src="/img/empty.svg"
    style="color: transparent;"
    width="300"
  />
  <h2
    class="text-2xl md:text-3xl font-semibold text-primary"
  >
    A simple title
  </h2>
  <p
    class="text-center text-lg font-light text-white"
  >
    Wanna play?
  </p>
</div>
`;
